### PR TITLE
Add relationship samples

### PR DESCRIPTION
**Pull request by azabbasi**
===RELATIONSHIP SAMPLE===

Created model dtmi:samples:Building;152530089
Created model dtmi:samples:Floor;157044262
Created twinbuildingTwinId74311011
Created twinfloorTwinId16629386

===CREATE RELATIONSHIPS===

Created a digital twin relationship buildingFloorRelationshipId from twin: buildingTwinId74311011 to twin: floorTwinId16629386

===GET RELATIONSHIP===

Retrieved relationship: buildingFloorRelationshipId from twin: buildingTwinId74311011
	Prop1: Prop1 value
	Prop2: 6

===LIST RELATIONSHIPS===

Retrieved relationship: buildingFloorRelationshipId with source: buildingTwinId74311011 and target: floorTwinId16629386

===LIST INCOMING RELATIONSHIPS===

Found an incoming relationship: buildingFloorRelationshipId from: buildingTwinId74311011

===DELETE RELATIONSHIP===

Deleted relationship: buildingFloorRelationshipId

Process finished with exit code 0

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        ConsoleLogger.printSuccess("Created a digital twin relationship "+ buildingFloorRelationshipId + " from twin: " + buildingTwinId + " to twin: " + floorTwinId);
+
+        ConsoleLogger.printHeader("Get Relationship");
+        Response<BasicRelationship> getRelationshipRepsonse = client.getRelationshipWithResponse(
+            buildingTwinId,
+            buildingFloorRelationshipId,
+            BasicRelationship.class,
+            Context.NONE);
+
+        if (getRelationshipRepsonse.getStatusCode() == HttpURLConnection.HTTP_OK) {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java:126]: Is this check required? Wouldn't something other than the accepted response code generate an exception?

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            .endpoint(parsedArguments.getDigitalTwinEndpoint())
+            .httpLogOptions(
+                new HttpLogOptions()
+                    .setLogLevel(parsedArguments.getHttpLogDetailLevel()))
+            .buildClient();
+
+        runRelationshipsSample();
+    }
+
+    @SuppressWarnings("rawtypes")
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java:52]: What does this do?

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java (hunk 1 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java:126]: not really a check but a sample code on how to check status codes ... we don't really need it for the sample to function

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java (hunk 2 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java:52]: needs to go

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/RelationshipsSyncSamples.java:52]: it allows us to use Response without any type parameters.

**Pull request merged**